docs/latest/design/egctl.md

```diff
@@ -0,0 +1,67 @@
+# Introduce egctl
+
+## Motivation
+
+EG should provide a command line tool with following capabilities:
+
+- One-step installation/uninstallation experience
+- Collect configuration from envoy proxy and gateway
+- Update admin options such as changing log level
+- Analyse system configuration to diagnose any issues in envoy gateway
+
+This tool is named `egctl`.
+
+## Syntax
+
+Use the following syntax to run `egctl` commands from your terminal window:
+
+```console
+egctl [command] [subcommand] [name] [flags]
+```
+
+where `command`, `name`, and `flags` are:
+
+* `command`: Specifies the operation that you want to perform on one or more resources,
+  for example `install`, `uninstall`, `config`, `version`.
+
+* `name`: Specifies the name of the specified pod. 
```

Comment:
Currently, EG only supports the Kubernetes provider. However, we do intend on supporting other providers, e.g. File. We should consider how egctl will support different providers. For example, when the `--provider` flag is unspecified, egctl will assume the Kubernetes provider. With that said, we should avoid Kube'isms such as `pod`.

Reply:
> using actual names (envoy-gateway, envoy-proxy) allows us to fetch info from specific resources

@arkodg can you provide an example to help me understand what you mean? Since egctl should interact with different EG instances, i.e. control/data planes, maybe we should consider supporting a `config` subcommand similar to `kubectl`. A user can set the credentials, EG endpoint, etc. to interact with different EG control/data planes.

Reply:
`egctl config envoy-proxy` gets envoy specific config (LDS, RDS etc)
`egctl config envoy-ratelimit-service` gets ratelimit service config

Reply:
I think it does make sense here to not explicitly call this `name` of a `pod`, since this project plans on supporting multiple providers, instead call it `name` of the runnable entity (for e.g. pod for the kubernetes provider)